src/components/Blog.js

```diff
@@ -0,0 +1,63 @@
+import React, {Component, Fragment} from 'react';
+import { Card, CardImg } from "react-bootstrap";
+import '../App.css';
+import { postsDb } from '../firebase';
+
+class BlogPage extends Component {
+
+        state = {
+            postsList: []
+        }
+
+    componentDidMount(){
+       postsDb.ref('posts').on("value", snapshot => {
+            let postlist = [];
+            snapshot.forEach(snap => {
```

Comment:
I'd consider using `.map()` here for this. `.forEach()` will iterate through the array, but it won't return anything. `.map()` will return an array based off each entry in the array. In other words, it maps one array to another. So here you could have something like this:

```javascript
postlist = snapshot.map(snap => snap.val());
```

Reply:
That’s correct! In here the the `.forEach()` won’t return anything. It iterates and pushes the `snap.val()` to the array. I applied the `.map()` in the `Card` to return the value. I considered it that way if I am right.